packages/block-library/src/search/variations.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const variations = [
+	{
+		name: 'default',
+		isDefault: true,
+		attributes: { buttonText: __( 'Search' ), label: __( 'Search' ) },
+	},
+];
```

Comment:
That's an interesting usage of "Variations" :)

Reply:
I tried the same approach with File block but it uses placeholders so it loses the attribute that is sourced from HTML after save and refresh because it isn't stored with `save` method. 

There are drawbacks :)

Reply:
A note (mostly to myself) for future reference. We could try to experiment with `supports: { placeholder: { type: 'media' } }`. With that flag, it would show media placeholder and when saving it would serialize attributes sources from HTML as attributes serialized in HTML comment of the block. It would resolve the issue with variations. The same could work with `supports: { placeholder: { type: 'variations' } }` for Columns block but it would show the variations picker when `save` returns null.